projects/templates/src/lib/components/po-page-dynamic-table/po-page-dynamic-table.component.ts

```diff
@@ -601,53 +605,94 @@ export class PoPageDynamicTableComponent extends PoPageDynamicListBaseComponent
 
     if (allow) {
       const uniqueKey = this.formatUniqueKey(item);
+      const resourceToRemoveKey = this.returnResourcesKeys([item]);
 
       if (typeof actionRemove === 'boolean' || newUrl) {
-        const isRemoveFromUI = true;
         return this.poPageDynamicService
           .deleteResource(uniqueKey, newUrl)
-          .pipe(map(() => this.removeFromUI(item, isRemoveFromUI)));
+          .pipe(map(() => this.removeFromUI(resourceToRemoveKey, this.literals.removeSuccessNotification)));
       }
-      return of(actionRemove(uniqueKey, item)).pipe(map(remove => this.removeFromUI(item, remove)));
+
+      return of(actionRemove(uniqueKey, item)).pipe(
+        tap(remove => {
+          const removeItem = remove ?? false;
+          this.removeFromUI(resourceToRemoveKey, this.literals.removeSuccessNotification, removeItem);
+        })
+      );
```

Comment:
Precisa mesmo do `of`, `pipe` e `tap`?
```suggestion
  const isRemove = actionRemove(uniqueKey, item) ?? false;
  this.removeFromUI(resourceToRemoveKey, this.literals.removeSuccessNotification, isRemove);
```
Pois a documentação da função espera que seja retornado um boolean, e ao colocar para retornar um observable, a principio parece não ter efeito.

Reply:
Eu preciso retornar um obs pois o switchMap esta esperando um obs.

Reply:
isso mesmo, o `switchMap` precisa, mas na parte inferior do método já está chamando um `of({})`, que parece cobrir essa situação, não daria para utilizá-lo?

Reply:
Conforme conversado, a diferença entre a sugestão e o jeito que está aparece na ordem de execução dos comandos, onde utilizando o obs fica toda sob responsabilidade do RxJS e no caso sugerido é executado de forma imperativa. As duas implementações funcionam para esse caso, mas se no futuro depois da emoção na tela eu queria encadear uma operação assincrono, utilizar o paradigma reativo vai deixar a função menos propensas a erros  de condição de corrida e outro erros de assincronicidade.